src/buffer.h

```diff
@@ -39,8 +39,12 @@ class Buffer {
   const DatumType& GetDatumType() const { return datum_type_; }
 
   void SetSize(size_t size) { size_ = size; }
-  size_t GetSize() const { return size_; }
-  size_t GetSizeInBytes() const { return size_ * datum_type_.SizeInBytes(); }
+  size_t GetSize() const {
+    if (size_ == 0 && !data_.empty())
```

Comment:
The test for data_.empty() is redundant. :-)

Reply:
Yea, removed.